include/fmt/os.h

```diff
@@ -396,8 +396,10 @@ class ostream final : private detail::buffer<char> {
   ostream(ostream&& other)
       : detail::buffer<char>(other.data(), other.size(), other.capacity()),
         file_(std::move(other.file_)) {
+    other.clear();
     other.set(nullptr, 0);
```

Comment:
I wonder if this can be replaced with

```
other = detail::buffer<char>();
```

Reply:
It seems reasonable, but I prefer the consistency between `set` and `clear`. 
Another issue is that creating another object in ctors seems confusing.